File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double maxnum = Double.NEGATIVE_INFINITY;
        double temp = 0;
        for(int i = 0; i < nums.length; i++){
            temp  += nums[i];
            if(i >= k){
                temp -= nums[i - k];
            }
            if(i >= k-1){
                maxnum = Math.max(maxnum , temp);
            }
        }
        return maxnum / k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxnum = float('-inf')
        temp = 0
        for i in range(len(nums)):
            temp += nums[i]
            if i >= k:
                 temp -= nums[i - k]
            if i >= k-1:
                maxnum = max(maxnum , temp)
        return maxnum/k